packages/ember-runtime/lib/system/core_object.js

```diff
@@ -418,7 +436,22 @@ CoreObject.PrototypeMixin = Mixin.create({
     @default false
     @public
   */
-  isDestroying: false,
+  isDestroying: descriptor({
+    get() {
+      return meta(this).sourceDestroying;
+    },
+
+    set(value) {
+      // prevent setting while applying mixins
```

Comment:
The guard here is to prevent an issue in our mixin apply code (we set the value before calling the `descriptor#setup` so the value being guarded against is this descriptor itself).

In general, I don't care about allowing this to be set outside of `destroy` and I can change this to throw if not being set by the quirk/bug in `defineProperty` mentioned above.